update-dependencies/CliDependencyHelper.cs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace Dotnet.Docker.Nightly
+{
+    public static class CliDependencyHelper
+    {
+        private static readonly Lazy<HttpClient> DownloadClient = new Lazy<HttpClient>();
+
+        public static string GetSharedFrameworkVersion(string cliVersion)
+        {
+            Trace.TraceInformation($"Looking for the Shared Framework CLI '{cliVersion}' depends on.");
+
+            string cliCommitHash = GetCommitHash(cliVersion);
+            XDocument depVersions = DownloadDependencyVersions(cliCommitHash).Result;
+            XNamespace msbuildNamespace = "http://schemas.microsoft.com/developer/msbuild/2003";
```

Comment:
I usually get the namespace of the `XDocument` for flexibility, and I might use `Descendants` below, but it's probably not needed here.